taxi/views.py

```diff
@@ -1,47 +1,55 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponse
 from django.shortcuts import render
 from django.views import generic
 
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
 
+    num_visists = request.session.get("num_visits", 0)
```

Comment:
There is a typo in the session variable name `num_visists`. It should be `num_visits` to match the rest of the code and ensure consistency.